### PR TITLE
fixed the nonce bug

### PR DESCRIPTION
**Pull request by nitzantomer**
there wasn't an option to not use the nonce, the previous code assumed that if no nonce was sent then the 'nonce' property wouldn't exist, but in reality it was there but with the value as undefined.
the tests did not catch it because the JWT service we used would add it's own nonce default value, that was fixed in the JWT service repo and the updated image is now used here.

### CODE REVIEW

File: scripts/src/models/orders.ts [hunk 1]
```diff
@@ -323,11 +323,15 @@ export type ExternalOrderFactory = OrderFactory & {
...
 function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
-		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
+		"new"(data?: Mutable<DeepPartial<Order>>, ...context: Array<DeepPartial<OrderContext>>): Order {
 			data = Object.assign(
-				{ nonce: Order.DEFAULT_NONCE },
 				data,
 				{ origin });
+
+			if (typeof data.nonce !== "string") {
```
Comment (doodyparizada) [inline scripts/src/models/orders.ts]: why wouldn't it be string?

Comment (doodyparizada) [reply scripts/src/models/orders.ts]: isNothing

File: package.json [hunk 2]
```diff
@@ -59,7 +59,7 @@
     "sqlite3": "^4.0.2",
     "tslint": "^5.9.1",
     "tslint-eslint-rules": "^4.1.1",
-    "typescript": "^2.7.2"
+    "typescript": "^2.8"
```
Comment (liorama) [inline package.json]: Please put this into its own commit or at least add a comment about this in the commit summary.

File: scripts/src/models/orders.ts (hunk 1 above)
Comment (liorama) [reply scripts/src/models/orders.ts]: I would check if the property exists vs checking it's value, eg `data.hasOwnProperty('nounce')` or `'nounce' in data` if you want to live on the edge (I think it'll be ok).

File: scripts/src/models/orders.ts [hunk 3]
```diff
@@ -325,9 +325,13 @@ function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
 		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
 			data = Object.assign(
-				{ nonce: Order.DEFAULT_NONCE },
 				data,
 				{ origin });
+
+			if (isNothing(data.nonce)) {
```
Comment (liorama) [inline scripts/src/models/orders.ts]: I would check if the property exists vs checking it's value, eg `data.hasOwnProperty('nounce')` or `'nounce' in data` if you want to live on the edge (I think it'll be ok). (cc @doodyparizada)

File: scripts/src/models/orders.ts [hunk 4]
```diff
@@ -325,9 +325,13 @@ function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
...
+				(data as Mutable<Order>).nonce = Order.DEFAULT_NONCE;
```
Comment (YohayBar) [inline scripts/src/models/orders.ts]: you need to make sure no one will alter Order.DEFAULT_NONCE in the future - might be an overkill but simple comment can help
